lib/inventory/controller.go

```diff
@@ -1293,10 +1355,24 @@ func (c *Controller) keepAliveDatabaseServer(handle *upstreamHandle, now time.Ti
 			srv.keepAliveErrs = 0
 			c.testEvent(dbKeepAliveOk)
 		}
-	} else if srv.retryUpsert {
-		srv.resource.SetExpiry(time.Now().Add(c.serverTTL).UTC())
+
+	} else if srv.retryUpsert || srv.heartbeatReservation != nil {
```

Comment:
Aren't we ignoring the new version of the resource here if `lease != nil`? Having a lease just means that we have upserted something in the past and we're just going to bump its keepalive without changing the value, right? Shouldn't the check for `heartbeatReservation != nil` happen before the check on `lease`?